graphql/schemabuilder/reflect.go

```diff
@@ -507,6 +518,9 @@ func (sb *schemaBuilder) buildFunction(typ reflect.Type, fun reflect.Value) (*gr
 			var result interface{}
 			if hasRet {
 				result = out[0].Interface()
+				if out[0].Kind() == reflect.Ptr && out[0].IsNil() && !m.MarkedNullable {
```

Comment:
I worry having non-pointer types will lead to people writing `return *foo` and causing panics.